server/config.go

```diff
@@ -421,21 +422,11 @@ func (c *Config) genEmbedEtcdConfig() (*embed.Config, error) {
 	return cfg, nil
 }
 
-// Generate a unique port for etcd usage. This is only used for test.
-// Because initializing etcd must assign certain address.
-func freePort() uint64 {
-	l, err := net.Listen("tcp", "127.0.0.1:0")
-	if err != nil {
-		panic(err)
-	}
-
-	port := uint64(l.Addr().(*net.TCPAddr).Port)
-	l.Close()
-
-	// wait a little to avoid using binding address.
-	time.Sleep(50 * time.Millisecond)
+var unixURLCount uint64
 
-	return port
+// unixURL returns a unique unix socket url, used for test only.
+func unixURL() string {
+	return fmt.Sprintf("unix://localhost:%d.%d.sock", os.Getpid(), atomic.AddUint64(&unixURLCount, 1))
```

Comment:
On my machine, they will be deleted automatically even if tests failed, however if we Interrupt tests, they will not be deleted.